public/app/features/teams/TeamGroupSync.tsx

```diff
@@ -36,7 +42,7 @@ const headerTooltip = `Sync LDAP or OAuth groups with your Grafana teams.`;
 export class TeamGroupSync extends PureComponent<Props, State> {
   constructor(props: Props) {
     super(props);
-    this.state = { isAdding: false, newGroupId: '' };
+    this.state = { isAdding: false, newGroupId: '', isReadOnly: props.isReadOnly };
```

Comment:
Also there's no need to sae it to the state, better read it directly from props.